internal/validators/hierarchy.go

```diff
@@ -250,11 +251,12 @@ func (v *Hierarchy) getConflictingObjectsOfType(gvk schema.GroupVersionKind, new
 	// Get all the source objects in the new ancestors that would be propagated
 	// into the descendants.
 	newAnsSrcObjs := make(map[string]bool)
-	// TODO additionally check if the ancestor source objects obey the
-	//  'shouldPropagateSource()' rules from the reconcilers/object.go. Only
-	//  propagatable ancestor source would cause overwriting conflict.
 	for _, o := range newParent.GetAncestorSourceObjects(gvk, "") {
-		newAnsSrcObjs[o.GetName()] = true
+		// If the user has chosen not to propagate the object to this descendant,
+		// then it should not be included in conflict checks
+		if ok, _ := selectors.ShouldPropagate(o, o.GetLabels()); ok {
```

Comment:
We should never throw away an error - instead, we need to propagate it up to the caller if need be. E.g.:

```
return fmt.Errorf("while determining if this change will conflict with %s %q from namespace %q: %w",
   gvk.Kind, o.GetName(), o.GetNamespace(), err)
```

Reply:
ShouldPropagate will always return false, even if there is an error, it is essentially replicating the missing conflict logic into hierarchy from object - https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/master/internal/validators/object.go#L335 and hncconfig https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/master/internal/validators/hncconfig.go#L169 validations

in these other cases the error is also dropped, as it does not make much to propagate it up to the caller

Reply:
Huh. Weird. You make a good point that there's a precedent for throwing the error away so I don't think we should hold up this PR for that reason.

I'm not thrilled about throwing away those error though - they can happen (e.g. if a bug creeps in) and we'll silently discard them. I'll file a separate bug for that.